vendor/github.com/prometheus/common/config/http_config.go

```diff
@@ -188,13 +189,13 @@ type HTTPClientConfig struct {
 	// Authorization.CredentialsFile.
 	BearerTokenFile string `yaml:"bearer_token_file,omitempty" json:"bearer_token_file,omitempty"`
 	// HTTP proxy server to use to connect to the targets.
-	ProxyURL URL `yaml:"proxy_url,omitempty" json:"proxy_url,omitempty"`
+	ProxyURL *URL `yaml:"proxy_url,omitempty" json:"proxy_url,omitempty"`
```

Comment:
We should be very carefull with this kind of fix here :
- it can be easily overridden
- from other part of the code, while it looks like we are using the official library, we are using a custom one

Reply:
May be we should add a a comment here to remind us that we did change it.

Reply:
I have added a test to ensure Marshal / Unmarshal works as expected until we fork / update promConfig